pkg/ruby/code_search.go

```diff
@@ -0,0 +1,179 @@
+package ruby
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/url"
+	"os"
+	"path"
+
+	"github.com/google/go-github/v58/github"
+
+	"github.com/wolfi-dev/wolfictl/pkg/gh"
+)
+
+// SearchResult represents Github Code Search results
+type SearchResult struct {
+	Repository string
+	FilePath   string
+	URL        string
+	Fragments  []string
+}
+
+// CodeSearch is the main search function which uses Github Code Search to run
+// the specified query for a particular package.
+func (o *Options) CodeSearch(pkg *Package, query string) error {
```

Comment:
```suggestion
func (o *Options) CodeSearch(ctx context.Context, pkg *Package, query string) error {
```

as mauren said, lets be good citizens and plumb context through 🙏

Reply:
Plumbed ✅